adapters/repos/db/vector/cache/cache.go

```diff
@@ -18,7 +18,14 @@ import (
 
 const DefaultDeletionInterval = 3 * time.Second
 
+type CacheMultiple[T any] interface {
```

Comment:
nitpicking: (feel free to ignore)

Perhaps `MultiCache` would be a better name?

Then the methods would be `GetMulti`, `PreloadMulti`, `PrefetchMulti` ?